backend/typescript/services/implementations/checkInService.ts

```diff
@@ -0,0 +1,164 @@
+    volunteerService: IVolunteerService | null = null,
+  ) {
+    this.emailService = emailService;
+    this.volunteerService = volunteerService;
+  }
+
+  async getAllCheckIns(): Promise<CheckInDTO[]> {
+    let checkInDtos: CheckInDTO[] = [];
+    try {
+      const checkIns = await CheckIn.findAll({
+        order: [["start_date", "ASC"]],
+      });
+      checkInDtos = checkIns.map((checkIn: CheckIn) => {
+        return {
+          id: String(checkIn.id),
+          startDate: checkIn.start_date,
+          endDate: checkIn.end_date,
+          notes: checkIn.notes,
+          volunteerId: String(checkIn.volunteer_id),
+          isAdmin: checkIn.is_admin,
+        };
+      });
+    } catch (error) {
+      Logger.error(
+        `Failed to get all checkins. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+
+    return checkInDtos;
+  }
+
+  async getCheckInsById(id: string): Promise<CheckInDTO> {
+    let checkIn: CheckIn | null;
+
+    try {
+      checkIn = await CheckIn.findByPk(Number(id));
+
+      if (!checkIn) {
+        throw new Error(`scheduling with id ${id} not found.`);
+      }
+    } catch (error) {
+      Logger.error(
+        `Failed to get check in by id. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+
+    return {
+      id: String(checkIn.id),
+      startDate: checkIn.start_date,
+      endDate: checkIn.end_date,
+      notes: checkIn.notes,
+      volunteerId: String(checkIn.volunteer_id),
+      isAdmin: checkIn.is_admin,
+    };
+  }
+
+  async getCheckInsByVolunteerId(volunteerId: string): Promise<CheckInDTO[]> {
+    let checkInDtos: CheckInDTO[] = [];
+    let checkIns: CheckIn[];
+    try {
+      checkIns = await CheckIn.findAll({
+        where: {
+          volunteer_id: Number(volunteerId),
+        },
+        order: [["start_date", "ASC"]],
+      });
+
+      checkInDtos = checkIns.map((checkIn) => {
+        return {
+          id: String(checkIn.id),
+          startDate: checkIn.start_date,
+          endDate: checkIn.end_date,
+          notes: checkIn.notes,
+          volunteerId: String(checkIn.volunteer_id),
+          isAdmin: checkIn.is_admin,
+        };
+      });
+    } catch (error) {
+      Logger.error(
+        `Failed to get schedules by volunteer id. Reason = ${getErrorMessage(
+          error,
+        )}`,
+      );
+      throw error;
+    }
+
+    return checkInDtos;
+  }
+
+  async deleteCheckInById(id: string): Promise<void> {
+    try {
+      const numDestroyed = await CheckIn.destroy({
+        where: { id: Number(id) },
+      });
+      if (numDestroyed <= 0) {
+        throw new Error(`checkin with id ${id} was not deleted.`);
+      }
+    } catch (error) {
+      Logger.error(
+        `Failed to delete checkin by id. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+  }
+
+  async deleteCheckInsByDateRange(
+    startDate: string,
+    endDate: string,
```

Comment:
I think we should probably validate that the startDate is before endDate?

Reply:
PR #153 includes this check in the validators :)

Reply:
ty for pointing it out :)